spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/OverwriteFiles.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.sql.catalyst.plans.logical
+
+import org.apache.spark.sql.catalyst.analysis.NamedRelation
+import org.apache.spark.sql.connector.write.BatchWrite
+
+case class OverwriteFiles(
```

Comment:
If we rename the builder to `ReplaceBuilder` then we'd probably want to call this `ReplaceData`. Replace because that's the operation we are using for the builder (since overwrite is part of a normal `Write`) and "data" because this isn't specific to files. It could also replace other groups, like whole partitions.